tribits/devtools_install/install-cmake.py

```diff
@@ -160,6 +163,8 @@ def doUntar(self):
     createDir(self.cmakeSrcDir, verbose=True)
     echoRunSysCmnd("tar -xzf "+self.cmakeTarball \
      +" -C "+self.cmakeSrcDir+" --strip-components 1")
+    echoRunSysCmnd("patch -d "+self.cmakeSrcDir+" -p1 -i " \
+     +os.path.join(devtools_install_dir, "0001-CTest-Add-CTEST_RESOURCE_SPEC_FILE-variable.patch"))
```

Comment:
I think we need an if-statement around this to only apply this patch if the version starts with "3.17".  For versions 3.18+, we will assume that the CMake source already has this patch.  For versions before 3.17, there will just be no support for CTEST_RESOURCE_SPEC_FILE.